File: Components/ASP.NET/CS/ASPxHtmlEditorDemos/App_Code/HtmlEditorUtils.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Web.UI;
using DevExpress.Web.ASPxHtmlEditor;

public static class HtmlEditorUtils {

    const string htmlLocation = "~/App_Data/Html";
    static readonly ICollection<string> darkThemeNames = new string[] { "BlackGlass", "PlasticBlue", "RedWine" };    

    public static void SetHtml(Page page, ASPxHtmlEditor htmlEditor, string htmlFileName) {
        string resolvedPath = page.MapPath(Path.Combine(htmlLocation, htmlFileName));
        htmlEditor.Html = File.ReadAllText(resolvedPath);
    }
    public static string GetHtml(Page page, string htmlFileName) {
        string resolvedPath = page.MapPath(Path.Combine(htmlLocation, htmlFileName));
        return File.ReadAllText(resolvedPath); 
    }
    public static string GetHtml(Page page, string htmlFileName, bool demoPageIsInRootFolder) {
        string result = GetHtml(page, htmlFileName);
        return demoPageIsInRootFolder ? result : result.Replace("UploadFiles/", "../UploadFiles/"); 
    }

    public static bool IsDarkTheme(string themeName) {
        return darkThemeNames.Contains(themeName);
    }

}

File: Components/ASP.NET/VB/ASPxHtmlEditorDemos/App_Code/HtmlEditorUtils.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Web.UI
Imports DevExpress.Web.ASPxHtmlEditor

Public NotInheritable Class HtmlEditorUtils

	Private Const htmlLocation As String = "~/App_Data/Html"
	Private Shared ReadOnly darkThemeNames As ICollection(Of String) = New String() { "BlackGlass", "PlasticBlue", "RedWine" }

	Private Sub New()
	End Sub
	Public Shared Sub SetHtml(ByVal page As Page, ByVal htmlEditor As ASPxHtmlEditor, ByVal htmlFileName As String)
		Dim resolvedPath As String = page.MapPath(Path.Combine(htmlLocation, htmlFileName))
		htmlEditor.Html = File.ReadAllText(resolvedPath)
	End Sub
	Public Shared Function GetHtml(ByVal page As Page, ByVal htmlFileName As String) As String
		Dim resolvedPath As String = page.MapPath(Path.Combine(htmlLocation, htmlFileName))
		Return File.ReadAllText(resolvedPath)
	End Function
	Public Shared Function GetHtml(ByVal page As Page, ByVal htmlFileName As String, ByVal demoPageIsInRootFolder As Boolean) As String
		Dim result As String = GetHtml(page, htmlFileName)
		Return If(demoPageIsInRootFolder, result, result.Replace("UploadFiles/", "../UploadFiles/"))
	End Function

	Public Shared Function IsDarkTheme(ByVal themeName As String) As Boolean
		Return darkThemeNames.Contains(themeName)
	End Function

End Class
